### PR TITLE
Use label encoding and embedding

### PR DESCRIPTION
**Pull request by yangtcai**
Currently, the implementation mAP still has some bugs, I will fix them ASAP.

### CODE REVIEW

File: configuration.yaml [hunk 1]
```diff
@@ -8,13 +8,14 @@ experiment_prefix: standard
 ################################################################################
 
 # dataset
-################################################################################
+###########################################ß#####################################
```
Comment (williamstark01) [inline configuration.yaml:11]: What is this German character doing here? :P

File: configuration.yaml [hunk 2]
```diff
@@ -8,13 +8,14 @@ experiment_prefix: standard
...
 #chromosomes: ['chr1', 'chr2', 'chrX']
 chromosomes: ['chr10']
 segment_length: 2000
 overlap: 500
 num_classes: 5 
-num_queries: 10      
+num_queries: 10
+num_nucleobase_letters: 6
```
Comment (williamstark01) [inline configuration.yaml:18]: Maybe I'm forgetting this one but what is the 6th nucleobase letter used in our dataset, other than ["A", "C", "G", "T", "N"]?

File: transformer.py [hunk 3]
```diff
@@ -378,7 +378,7 @@ def _get_activation_fn(activation):
 
 
 def build_transformer(configuration):
-    return Transformer(d_model=6, nhead=6, dropout=configuration.dropout)
+    return Transformer(d_model=32, nhead=8, dropout=configuration.dropout)
```
Comment (williamstark01) [inline transformer.py:381]: How about converting d_model (better called "embedding_dimension") and number of heads to variables in the config file? Also, I think it's better to delete this function completely, as it will be clearer to simply use the Transformer class directly to create an instance of it anywhere it's needed.

**Review by williamstark01 — CHANGES_REQUESTED** [review]
Good job, just a couple small things to fix. I'll start some training experiments myself to see how it goes.

File: configuration.yaml (hunk 2 above)
Comment (yangtcai) [reply configuration.yaml:18]: Those codes were borrowed from your previous [project](https://github.com/Ensembl/gene_pcp), I think the reason why the `num_nucleobase_letters` is 6 is that there is a padding letter.
https://github.com/yangtcai/Ensembl-Repeat-Identification/blob/0410964aa2ee88759d5f0776c2bcd260954f9eaf/dataloader.py#L251-L258

**Review by williamstark01 — CHANGES_REQUESTED** [review]
Good job, just one last tiny thing to update.

closes #35
closes #42

File: model.py [hunk 4]
```diff
@@ -18,7 +18,7 @@ class DETR(nn.Module):
     Copy-paste from DETR module with modifications
     """
 
-    def __init__(self, transformer, num_classes, num_queries):
+    def __init__(self, transformer, num_classes, num_queries, num_nucleobase_letters):
```
Comment (yangtcai) [inline model.py:21]: the `num_nucleobase_letters ` in configuration will be passed in here

File: dataloader.py [hunk 5]
```diff
@@ -255,9 +257,8 @@ class DnaSequenceMapper:
 
     def __init__(self):
         nucleobase_symbols = ["A", "C", "G", "T", "N"]
-        padding_character = [" "]
 
-        self.nucleobase_letters = sorted(nucleobase_symbols + padding_character)
+        self.nucleobase_letters = sorted(nucleobase_symbols)
 
         self.num_nucleobase_letters = len(self.nucleobase_letters)
```
[the remark this answers is not in the document]
Comment (williamstark01) [reply dataloader.py:263]: Not an excellent way and feels somewhat hacky, but in another project I simply saved the variable in the configuration object which is afterwards used to create the network instance:

https://github.com/Ensembl/gene_pcp/blob/main/utils.py#L247-L249

https://github.com/Ensembl/gene_pcp/blob/main/transformer_pipeline.py#L448-L456

https://github.com/Ensembl/gene_pcp/blob/main/transformer_pipeline.py#L217

I think it will work for this project as well.

**Review by williamstark01 — APPROVED** [review]
Everything looks good, onward to the next goal!
